src/Misc/layoutroot/run.cmd

```diff
@@ -24,10 +24,10 @@ if /i "%~1" equ "localRun" (
   rem ********************************************************************************
   "%~dp0bin\Runner.Listener.exe" run %*
 
-  rem Return code 4 means the run once runner received an update message.
+  rem Return codes 3 and 4 mean the runner received an update message.
   rem Sleep 5 seconds to wait for the update process finish and run the runner again.
-  if ERRORLEVEL 4 (
+  if ERRORLEVEL 3 (
```

Comment:
@TingluoHuang can i get your eyes on this. It looks like a bug but all related history goes back to the initial squash so i'm confused why `if ERRORLEVEL4` works (see const `RunnerUpdating`)

Reply:
@fhammerl and i looked at this together

Reply:
3 is a regular update, 4 is run once agent update.

Reply:
@TingluoHuang why sometimes does this script restart the runner, and sometimes the runner is restarted by `start "Actions Runner" cmd.exe /k "_ROOT_FOLDER_\run.cmd"` in `src/Misc/layoutbin/update.cmd.template`?

Reply:
@MatisseHack curious is `start` what is causing the issue you are seeing? if so then based on your proposed fix, it looks like `start` command would never be invoked anymore (if we go that route, we should delete it)

Reply:
i guess `start` inherits the user profile from the logged in session, not from the process?

Reply:
https://github.com/actions/runner/pull/1124#discussion_r641764098

Reply:
Replied above, but unfortunately removing `START` was not the fix. I did narrow down the issue to C#'s method of creating a new process.